MSAL/test/app/ios/MSALTestAppAuthorityViewController.m

```diff
@@ -50,35 +49,46 @@ + (instancetype)sharedController
     return s_controller;
 }
 
-- (id)init
+- (instancetype)init
 {
-    if (!(self = [super init]))
-    {
-        return nil;
-    }
-    
-    NSString *currentAuthority = MSALTestAppSettings.settings.authority;
-    _authorities = [[MSALTestAppSettings authorities] mutableCopy];
-    if (currentAuthority && ![_authorities containsObject:currentAuthority])
-    {
-        [_authorities addObject:currentAuthority];
-    }
-
-    _savedAuthorities = [[[NSUserDefaults standardUserDefaults] objectForKey:@"saved_authorities"] mutableCopy];
+    return [self initWithAuthorities:[[MSALTestAppSettings authorities] mutableCopy]
+                     addUserDefaultsForKey:@"saved_aadAuthorities"];
+}
 
-    if (!_savedAuthorities)
+- (instancetype) initWithAuthorities:(NSMutableArray *)authorities
+                     addUserDefaultsForKey:(NSString *)key
```

Comment:
nit formatting notes to keep in mind in the future:

1. Don't add space after method return type, so should be:

```
- (instancetype)initWithAuthorities:(NSMutableArray *)authorities
                   addUserDefaultsForKey:(NSString *)key
```
2. Multiple ":"-s should preferably be aligned. But you'll notice that when doing refactoring using Xcode for multiple methods, it breaks alignment :(
3. Add a space after if and for, so something like:

```
if (self)
{
     ....
}
```

```
for (NSString *part in parts)
{
     ....
}
```

Reply:
done